src/cabal

```diff
@@ -108,6 +109,23 @@ run_repl () {
 }
 
 #
+# Upgrade the script in place
+#
+run_upgrade () {
+    echo "Checking for a new version of cabal extras ..."
+    TZ=$(date +"%T")
+    curl --silent "https://raw.githubusercontent.com/ambiata/cabal-extras/master/src/cabal?$TZ" > /tmp/cabal
+    if ! cmp ./cabal /tmp/cabal  >/dev/null 2>&1
+    then
+      echo "New version found and upgraded. You can now commit it to your git repo."
+      mv /tmp/cabal ./cabal
```

Comment:
@charleso yeah that works - but not sure how to shoehorn the commit sha into the script itself (given it won't be known until merged into cabal-extras)